### PR TITLE
feat(client): ts-migrate client/src/components/profile

### PR DESCRIPTION
**Pull request by Dripcoding**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Migrating the following files to Typescript:
/client/src/components/profile/Profile.js
/client/src/components/profile/Profile.test.js
/client/src/components/profile/components/Camper.js
/client/src/components/profile/components/Certifications.js
/client/src/components/profile/components/HeatMap.js
/client/src/components/profile/components/HeatMap.test.js
/client/src/components/profile/components/Portfolio.js
/client/src/components/profile/components/SocialIcons.js
/client/src/components/profile/components/TimeLine.js
/client/src/components/profile/components/TimeLine.test.js
/client/src/components/profile/components/TimelinePagination.js

Ref: #42256

### CODE REVIEW

**Review by raisedadead — COMMENTED** [review]
Hey there,

Thank you for taking the time to contribute to the TypeScript migration. 

The main issue has been updated a lot over the last few days, and so we suggest you re-read it and update this pull request to match if you have the time.

The main goals for a PR are:
- [ ] Changing the filename from `PascalCase` to `kebab-case` and ensuring that the imports in code are all still valid.
- [ ] Changing the file extension to .tsx (if a file has React JSX syntax)
- [ ] Fixing as many TypeScript and Eslint errors as you can while refactoring the code.

If you need any assistance, please reach out in our [contributing chat room](https://chat.freecodecamp.org/channel/contributors).

File: client/src/components/helpers/Spacer.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    <Comp />
+  ) : (
+    <>
+      {'#'
+        .repeat(size)
+        .split('')
+        .map((_, i) => (
+          <Comp key={`spacer_${i}`} />
+        ))}
+    </>
```
Comment (ShaunSHamilton) [inline client/src/components/helpers/Spacer.tsx:24]:
```suggestion
  (
  <>
    {Array.from(Array(size), i => (
      <Comp key={`spacer_${i}`} />
    ))}
  </>
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Hello there. Thank you, for taking the time to contribute.

<s>This is a suggestion _stolen_ from @ojeytonwilliams in Chat. My thinking is to prioritise this PR in the merge queue so we can rebase the other PRs with this change.</s>

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
There should be no changes to `client/src/components/helpers/*` files. These changes need to be reverted.

**Comment by Dripcoding** [comment]
Thanks for the feedback! I'll work on up updating this pr for the party.

**Review by ShaunSHamilton — APPROVED** [review]
This LGTM 👍

**Pull request merged**